cns/restserver/ipam.go

```diff
@@ -434,26 +563,46 @@ func (service *HTTPRestService) unassignIPConfig(ipconfig cns.IPConfigurationSta
 func (service *HTTPRestService) releaseIPConfig(podInfo cns.PodInfo) error {
 	service.Lock()
 	defer service.Unlock()
-
-	ipID := service.PodIPIDByPodInterfaceKey[podInfo.Key()]
-	if ipID != "" {
-		if ipconfig, isExist := service.PodIPConfigState[ipID]; isExist {
-			logger.Printf("[releaseIPConfig] Releasing IP %+v for pod %+v", ipconfig.IPAddress, podInfo)
-			_, err := service.unassignIPConfig(ipconfig, podInfo)
-			if err != nil {
-				return fmt.Errorf("[releaseIPConfig] failed to mark IPConfig [%+v] as Available. err: %v", ipconfig, err)
+	numIPsToBeReleased := len(service.PodIPIDByPodInterfaceKey[podInfo.Key()])
+	ipsToBeReleased := make([]cns.IPConfigurationStatus, numIPsToBeReleased)
+
+	for i, ipID := range service.PodIPIDByPodInterfaceKey[podInfo.Key()] {
+		if ipID != "" {
+			if ipconfig, isExist := service.PodIPConfigState[ipID]; isExist {
+				ipsToBeReleased[i] = ipconfig
```

Comment:
you cannot use index i to add values to ipsToBeReleased. 
e.g.
i =0 -> you added hat to ipsToBeReleased.
i = 1 -> you hit the else so didn't add to ipsToBeReleased
i = 2 -> you see it exists : this will end up adding that value at i = 2.
so your ipsToBeReleased will have valid values at 0th and 2nd index.
Just append to ipsToBeReleased. which means no need to make it with upfront with numIPsTobeReleased on 567.

Reply:
That's a good point. I have been trying to avoid appending when I can but it just makes more sense here.